### PR TITLE
dev: nudge the cache if it is not running during `build`

### LINKED ISSUE #76527: dev,bazel: check if the cache process is running
**Describe the problem**

We recently rolled out `dev cache` to set up a locally running instance of `bazel-remote`. We also check whether the cache is up and running as part of `dev doctor`. Post reboot however, if running any `bazel`/`dev` commands, you'll observer the following warnings:

```
WARNING: Writing to Remote Cache:
13 errors during bulk transfer
WARNING: Writing to Remote Cache:
22 errors during bulk transfer
```

**Expected behavior**

Probably we should do a cheap check for the cache process, to see if it's running, and maybe kicking it if it isn't. At the very least suggest a manual `dev {setup,cache,doctor}`.

### PR DESCRIPTION
**Pull request by aayushshah15**
Release note: None

Jira issue: CRDB-14795

### CODE REVIEW

File: pkg/cmd/dev/build.go [hunk 1]
```diff
@@ -114,6 +114,18 @@ func (d *dev) build(cmd *cobra.Command, commandLine []string) error {
 	ctx := cmd.Context()
 	cross := mustGetFlagString(cmd, crossFlag)
 
+	bazelRcLine, err := d.setUpCache(ctx)
```
Comment (rickystewart) [inline pkg/cmd/dev/build.go:117]: In `doctor` we check the variable `DEV_NO_REMOTE_CACHE` and skip setting up the cache if it's set. Can you do the same here?

**Review by rickystewart — CHANGES_REQUESTED** [review]

File: pkg/cmd/dev/build.go (hunk 1 above)
Comment (aayushshah15) [reply pkg/cmd/dev/build.go]: Done.

File: pkg/cmd/dev/cache.go [hunk 2]
```diff
@@ -131,6 +131,7 @@ func (d *dev) cacheIsUp(ctx context.Context) bool {
 // string which is a line that should be added to ~/.bazelrc.
 func (d *dev) setUpCache(ctx context.Context) (string, error) {
 	if d.cacheIsUp(ctx) {
+		log.Printf("Cache is already up\n")
```
Comment (rickystewart) [inline pkg/cmd/dev/cache.go:134]: This seems like a noisy thing to print on every `dev` invocation. Can you remove this line?

**Review by rickystewart — APPROVED** [review]

File: pkg/cmd/dev/cache.go (hunk 2 above)
Comment (aayushshah15) [reply pkg/cmd/dev/cache.go]: Done.

**Comment by aayushshah15** [comment]
TFTR! 

bors r+

**Pull request merged**